src/Type.php

```diff
@@ -612,4 +621,24 @@ public function patchEntities($entities, array $data, array $options = [])
         $marshaller = $this->marshaller();
         return $marshaller->mergeMany($entity, $data, $options);
     }
+
+    /**
+     * Get the mapping data from the index type.
+     *
+     * This will fetch the schema from ElasticSearch the first
+     * time this method is called.
+     *
+     *
+     * @return array
+     */
+    public function schema()
+    {
+        if ($this->_schema !== null) {
+            return $this->_schema;
+        }
+        $type = $this->connection()->getIndex()->getType($this->name());
+        $mapping = $type->getMapping();
+        $this->_schema = $mapping[$this->name()]['properties'];
+        return $this->_schema;
```

Comment:
Ok, I can wrap it in a Table\Schema like object. Probably less fancy but similar interface for getting field names and field types. It will also make the nested field situation better.